test/e2e/providers/azure/azure.go

```diff
@@ -119,11 +124,218 @@ func (p *Provider) GetType() config.PlatformType {
 }
 
 func (p *Provider) StorageSupport() bool {
-	return false
+	return true
 }
 
-func (p *Provider) CreatePVC(_ client.Interface, _ string, _ *core.PersistentVolume) (*core.PersistentVolumeClaim, error) {
-	return nil, fmt.Errorf("storage not supported on azure")
+func (p *Provider) CreatePVC(c client.Interface, namespace string, _ *core.PersistentVolume) (*core.PersistentVolumeClaim, error) {
+	if err := p.ensureWindowsCSIDaemonSet(c); err != nil {
+		return nil, err
+	}
+	storageClassName := "azurefile-csi"
+	pvcSpec := core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: "e2e" + "-",
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: []core.PersistentVolumeAccessMode{core.ReadWriteMany},
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{core.ResourceStorage: resource.MustParse("2Gi")},
+			},
+			StorageClassName: &storageClassName,
+		},
+	}
+	return c.CoreV1().PersistentVolumeClaims(namespace).Create(context.TODO(), &pvcSpec, meta.CreateOptions{})
+}
+
+// ensureWindowsCSIDaemonSet deploys the Windows CSI driver DaemonSet if it doesn't already exist
+func (p *Provider) ensureWindowsCSIDaemonSet(client client.Interface) error {
+	dsName := "azure-file-csi-driver-node-windows"
+	directoryType := core.HostPathDirectory
+	directoryOrCreate := core.HostPathDirectoryOrCreate
+	ds := apps.DaemonSet{
+		ObjectMeta: meta.ObjectMeta{
+			Name: dsName,
+		},
+		Spec: apps.DaemonSetSpec{
+			Selector: &meta.LabelSelector{MatchLabels: map[string]string{"app": dsName}},
+			Template: core.PodTemplateSpec{
+				ObjectMeta: meta.ObjectMeta{
+					Labels: map[string]string{"app": dsName},
+				},
+				Spec: core.PodSpec{
+					PriorityClassName: "system-node-critical",
+					NodeSelector:      map[string]string{core.LabelOSStable: "windows"},
+					// Use the controller-sa as the node-sa doesn't have GET secrets permissions required for Windows
+					ServiceAccountName: "azure-file-csi-driver-controller-sa",
+					OS:                 &core.PodOS{Name: core.Windows},
+					Tolerations: []core.Toleration{
+						{
+							Key:    "os",
+							Value:  "Windows",
+							Effect: core.TaintEffectNoSchedule,
+						},
+					},
+					Containers: []core.Container{
+						{
+							Name:  "node-driver-registrar",
+							Image: "mcr.microsoft.com/oss/kubernetes-csi/csi-node-driver-registrar:v2.8.0",
```

Comment:
Why not using the [latest](https://github.com/kubernetes-csi/node-driver-registrar/releases/tag/v2.9.0) (v2.9.0) as of now?

Reply:
This is following the documentation provided https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/deploy